javalanglib/src/main/scala/java/lang/StackTrace.scala

```diff
@@ -188,7 +188,8 @@ private[lang] object StackTrace {
    */
   private def extractClassMethod(functionName: String): js.Array[String] = {
     val PatC = """^(?:Object\.|\[object Object\]\.)?\$c_([^\.]+)(?:\.prototype)?\.([^\.]+)$""".re
-    val PatS = """^(?:Object\.|\[object Object\]\.)?\$(?:ct|ps?|s|f)_((?:_[^_]|[^_])+)__([^\.]+)$""".re
+    val PatS = """^(?:Object\.|\[object Object\]\.)?\$(?:ps?|s|f)_((?:_[^_]|[^_])+)__([^\.]+)$""".re
+    val PatCT = """^(?:Object\.|\[object Object\]\.)?\$ct_((?:_[^_]|[^_])+)__([^\.]*)$""".re
```

Comment:
The encoding of `init` methods slightly changes in this PR, because the `init___` part of it disappeared (but we gained a trailing `__` in exchange), so `$ct_xyz` functions do not have the same format as `$p_`, `$s_` etc. functions. The actual difference in the regexp is the very last `*` instead of a `+`.